thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java

```diff
@@ -507,4 +523,159 @@ public Response evaluateAlertFilterByAutoTuneId(@PathParam("autotuneId") long id
     return Response.ok(evaluator.toProperties().toString()).build();
   }
 
+  /**
+   * Perform anomaly function autotune:
+   *  - run backfill on all possible combinations of tuning parameters
+   *  - keep all the parameter combinations which lie in the goal range
+   *  - return list of parameter combinations along with their performance evaluation
+   * @param functionId
+   * the id of the target anomaly function
+   * @param replayStartTimeIso
+   * the start time of the anomaly function replay in ISO format, e.g. 2017-02-27T00:00:00.000Z
+   * @param replayEndTimeIso
+   * the end time of the anomaly function replay in ISO format
+   * @param speedup
+   * whether we speedup the replay process
+   * @param tuningJSON
+   * the json object includes all tuning fields and list of parameters
+   * ex: {"baselineLift": [0.9, 0.95, 1, 1.05, 1.1], "baselineSeasonalPeriod": [2, 3, 4]}
+   * @param goal
+   * the expected performance assigned by user
+   * @return
+   * A response containing all satisfied properties with their evaluation result
+   */
+  @POST
+  @Path("function/{id}/replay")
+  public Response anomalyFunctionReplay(@PathParam("id") long functionId, @QueryParam("start") String replayStartTimeIso,
+      @QueryParam("end") String replayEndTimeIso, @QueryParam("speedup") @DefaultValue("true") boolean speedup,
+      @QueryParam("tune") String tuningJSON, @QueryParam("goal") double goal,
+      @QueryParam("evalMethod") @DefaultValue("ANOMALY_PERCENTAGE") String performanceEvaluationMethod){
+    DateTime replayStart = null;
+    DateTime replayEnd = null;
+    try {
+      replayStart = ISODateTimeFormat.dateTimeParser().parseDateTime(replayStartTimeIso);
+      replayEnd = ISODateTimeFormat.dateTimeParser().parseDateTime(replayEndTimeIso);
+    }
+    catch (Exception e) {
+      throw new WebApplicationException("Unable to parse strings, "+ replayStartTimeIso + " and " + replayEndTimeIso +
+          ", in ISO DateTime format", e);
+    }
+
+    // List all tuning parameter sets
+    List<Map<String, String>> tuningParameters = null;
+    try {
+      tuningParameters = listAllTuningParameters(new JSONObject(tuningJSON));
+    }
+    catch(JSONException e){
+      LOG.error("Unable to parse json string: {}", tuningJSON, e );
+      return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+    if(tuningParameters.size() == 0) { // no tuning combinations
+      LOG.warn("No tuning parameter is found in json string {}", tuningJSON);
+      return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+    AutotuneMethodType autotuneMethodType = AutotuneMethodType.EXHAUSTIVE;
+    PerformanceEvaluationMethod performanceEvalMethod = PerformanceEvaluationMethod.valueOf(performanceEvaluationMethod.toUpperCase());
+
+    // select the functionAutotuneConfigDTO in DB
+    Long targetId = null;
```

Comment:
This can be a TODO: override existing autotune results by a method "autotuneConfigDAO.udpate()"  Leave it in the future. :)